feature-flags-tmp/feature_flags/feature_flags.py

```diff
@@ -0,0 +1,222 @@
+import copy
+import re
+
+from fastapi import HTTPException
+from typing import Optional
+
+import reflex as rx
+from sqlmodel import Field
+
+
+class FeatureFlags(rx.Model, table=True):
+    name: str = Field(primary_key=True)
+    value: str = Field()
+
+
+class FeatureFlagsState(rx.State):
+    feature_flags_as_loaded_from_db: Optional[dict[str, str]] = None
+
+    pending_creates_or_updates: dict[str, str] = {}
+    pending_deletes: set[str] = set()
+
+    new_flag_modal_is_open: bool = False
+    new_flag_modal_error: Optional[str] = None
+
+    new_flag_modal_flag_name: Optional[str] = ""
+    new_flag_modal_flag_value: Optional[str] = ""
```

Comment:
i think the modal-related stuff should go into a substate, then you also wouldn't need `reset_new_flag_modal_state`, since you could just call `self.reset()` and use the builtin method if the substate only contained items related to the modal.